disaggregated_memory/configs/riscv-main.py

```diff
@@ -0,0 +1,293 @@
+)
+
+# Parameters related to local memory
+parser.add_argument(
+    "--local-memory-size",
+    type=str,
+    required=True,
+    help="Local memory size",
+)
+
+# Parameters related to remote memory
+parser.add_argument(
+    "--is-composable",
+    type=str,
+    required=True,
+    choices=["True", "False"],
+    help="Tell the simulation to either use gem5 or SST as the remote memory.",
+)
+parser.add_argument(
+    "--remote-memory-addr-range",
+    type=str,
+    required=True,
+    help="Remote memory range",
+)
+parser.add_argument(
+    "--remote-memory-latency",
+    type=int,
+    required=True,
+    help="Remote memory latency in Ticks (has to be converted prior)",
+)
+
+# Parameters related to checkpoints.
+parser.add_argument(
+    "--ckpt-file",
+    type=str,
+    default="",
+    required=False,
+    help="optionally put a path to restore a checkpoint",
+)
+parser.add_argument(
+    "--take-ckpt",
+    type=str,
+    default="False",
+    required=True,
+    help="optionally put a path to restore a checkpoint",
+)
+args = parser.parse_args()
+cpu_type = {
+    "o3": CPUTypes.O3,
+    "atomic": CPUTypes.ATOMIC,
+    "timing": CPUTypes.TIMING,
+    "kvm": CPUTypes.KVM,
+}[args.cpu_type]
+use_sst = {"True": True, "False": False}[args.is_composable]
+
+remote_memory_range = list(map(int, args.remote_memory_addr_range.split(",")))
+remote_memory_range = AddrRange(remote_memory_range[0], remote_memory_range[1])
+
+# This runs a check to ensure the gem5 binary is compiled for ARM.
+requires(isa_required=ISA.RISCV)
+# Here we setup the parameters of the l1 and l2 caches.
+cache_hierarchy = ClassicPrivateL1PrivateL2SharedL3DMCache(
+    l1d_size="32KiB", l1i_size="32KiB", l2_size="256KiB", l3_size="4MiB"
+)
+
+# Memory: Dual Channel DDR4 2400 DRAM device.
+local_memory = DualChannelDDR4_2400(size=args.local_memory_size)
+
+# Either suppy the size of the remote memory or the address range of the
+# remote memory. Since this is inside the external memory, it does not matter
+# what type of memory is being simulated. This can either be initialized with
+# a size or a memory address range, which is mroe flexible. Adding remote
+# memory latency automatically adds a non-coherent crossbar to simulate latenyc
+remote_memory = ExternalRemoteMemoryV2(
+    addr_range=remote_memory_range, use_sst_sim=use_sst
+)
+
+# Here we setup the processor. We use a simple processor.
+processor = SimpleProcessor(cpu_type=cpu_type, isa=ISA.RISCV, num_cores=1)
+
+# Here we setup the board which allows us to do Full-System ARM simulations.
+board = RiscvComposableMemoryBoard(
+    clk_freq=args.cpu_clock_rate,
+    processor=processor,
+    local_memory=local_memory,
+    remote_memory=remote_memory,
+    cache_hierarchy=cache_hierarchy,
+)
+
+# commands to execute to run the simulation.
+mount_cmd = ["mount -t sysfs - /sys;", "mount -t proc - /proc;"]
+
+warn("The command list to execute has to be manually set!")
```

Comment:
BTW, this is the whole point of the "workload" objects!

Reply:
We can package all the experiments as workload objects. I can spend some time to see how we can use it and customize switch CPUs.